src/meteors/attr/lime.py

```diff
@@ -1027,13 +1046,15 @@ def get_spatial_attributes(
             hsi (list[HSI] | HSI): Input hyperspectral image(s) for which the attributions are to be computed.
                 If a list of HSI objects is provided, the attributions are computed for each HSI object in the list.
                 The output will be a list of HSISpatialAttributes objects.
-            segmentation_mask (np.ndarray | torch.Tensor | None, optional):
+            segmentation_mask (np.ndarray | torch.Tensor | list[np.ndarray | torch.Tensor] | None, optional):
                 A segmentation mask according to which the attribution should be performed.
-                The segmentation mask should have a 3D shape, which can be broadcastable to the shape of the input
-                image. The only dimension on which the image and the mask shapes can differ is the spectral dimension,
-                marked with letter `C` in the `image.orientation` parameter. If None, a new segmentation mask is
-                created using the `segmentation_method`. Additional parameters for the segmentation method may be
-                passed as kwargs. Defaults to None.
+                The segmentation mask should have a 2D or 3D shape, which can be broadcastable to the shape of the
+                input image. The only dimension on which the image and the mask shapes can differ is the spectral
+                dimension, marked with letter `C` in the `image.orientation` parameter. If None, a new segmentation mask
+                is created using the `segmentation_method`. Additional parameters for the segmentation method may be
+                passed as kwargs. If multiple HSI images are provided, a list of segmentation masks can be provided,
+                one for each image. If list is not provided method will assume that the same segmentation mask is used
```

Comment:
well, this might not be the best approach, and I would rather create multiple segmentation masks, but I am fine with this approach now